### PR TITLE
Helpers for ReadyToRun

### PR DESCRIPTION
**Pull request by rahku**
- Implementation of missing helpers for arm64.
- indentation fixes for my previous checkin

### CODE REVIEW

**Comment by rahku** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by rahku** [comment]
@jkotas @dotnet/arm64-contrib PTAL

**Comment by jkotas** [comment]
LGTM. Does the ReadyToRun unit test `tests\src\readytorun` work now?

File: src/vm/arm64/stubs.cpp [hunk 1]
```diff
@@ -1709,53 +1709,283 @@ void StubLinkerCPU::EmitUnboxMethodStub(MethodDesc *pMD)
...
+    BEGIN_DYNAMIC_HELPER_EMIT(32);
+    
+    // mov x1, x0
+    *(DWORD*)p = 0x91000001;
+    p += 4;
+
+    // adr x8, <label>
+    // ldp x0, x12, [x8]
+    LoadRegPair(p, 0, 12, 12);
+	p += 8;
```
Comment (janvorli) [inline src/vm/arm64/stubs.cpp]: A nit - you seem to have tabs instead of spaces at some places

**Comment by rahku** [comment]
No I get a bunch of InvalidProgramException when generating NI. I have opened jit issue for that. But the ni gets generated successfully. On execution using the ni I get AV at below callstack:
CLRStub[MethodCallThunk]@7ffeeb2d2584
mainv1_ni!Program.Main()+0x44
CoreCLR!CallDescrWorkerInternal+0x7c
CoreCLR!CallDescrWorkerWithHandler+0x26c
CoreCLR!MethodDescCallSite::CallTargetWorker+0x1064
CoreCLR!MethodDescCallSite::Call_RetArgSlot+0xe8
CoreCLR!``RunMain'::`30'::__Body::Run'::`5'::__Body::Run+0x28c
CoreCLR!`RunMain'::`30'::__Body::Run+0x88
CoreCLR!RunMain+0x198
CoreCLR!Assembly::ExecuteMainMethod+0x42c
CoreCLR!CorHost2::ExecuteAssembly+0x56c
CoreRun!TryRun+0xc48
CoreRun!wmain+0x158
CoreRun!invoke_main+0x38
CoreRun!__scrt_common_main_seh+0x194
CoreRun!__scrt_common_main+0x10
CoreRun!wmainCRTStartup+0xc
KERNEL32!BaseThreadInitThunk+0x30
ntdll!RtlUserThreadStart+0x44
-----------------------------------------------------

InvalidProgramException while compiling method MyClass.TestStaticFields
ReadyToRun: embedGenericSignature not yet supported
Not implemented (Exception from HRESULT: 0x80004001 (E_NOTIMPL)) while compiling method MyGeneric`2[__Canon,__Canon][System.__Canon,System.__Canon].GenericVirtualMethod
ReadyToRun: embedGenericSignature not yet supported
Not implemented (Exception from HRESULT: 0x80004001 (E_NOTIMPL)) while compiling method MyGeneric`2[__Canon,__Canon][System.__Canon,System.__Canon].ChangedToVirtual
ReadyToRun: embedGenericSignature not yet supported
Not implemented (Exception from HRESULT: 0x80004001 (E_NOTIMPL)) while compiling method MyChildGeneric`1[__Canon][System.__Canon].MovedToBaseClass
ReadyToRun: embedGenericSignature not yet supported
Not implemented (Exception from HRESULT: 0x80004001 (E_NOTIMPL)) while compiling method MyChildGeneric`1[__Canon][System.__Canon].MovedToBaseClass
ReadyToRun: embedGenericSignature not yet supported
Not implemented (Exception from HRESULT: 0x80004001 (E_NOTIMPL)) while compiling method MyGeneric`2[__Canon,__Canon][System.__Canon,System.__Canon].GenericVirtualMethod
ReadyToRun: embedGenericSignature not yet supported
Not implemented (Exception from HRESULT: 0x80004001 (E_NOTIMPL)) while compiling method MyGeneric`2[__Canon,__Canon][System.__Canon,System.__Canon].ChangedToVirtual
Native image test.ni.dll generated successfully.
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.22220.0
Copyright (c) Microsoft Corporation.  All rights reserved.

ReadyToRun: Implicit boxing for calls to constrained methods not supported
Not implemented (Exception from HRESULT: 0x80004001 (E_NOTIMPL)) while compiling method Program.TestConstrainedMethodCalls_Unsupported
InvalidProgramException while compiling method Program.TestStaticFields
Native image mainv1.ni.exe generated successfully.

**Pull request merged**